geth/peers/peerpool.go

```diff
@@ -222,37 +229,64 @@ func (p *PeerPool) handleServerPeers(server *p2p.Server, events <-chan *p2p.Peer
 				retryDiscv5 = time.After(discoveryRestartTimeout)
 				log.Error("starting discv5 failed", "error", err, "retry", discoveryRestartTimeout)
 			}
+		case <-stopDiscv5:
+			p.handleStopTopics(server)
+			SendDiscoverySummary(server.PeersInfo())
```

Comment:
Since stopping search happens asynchronously (and also the change on number of peers) i moved `SendDiscoverySummary` where this change on the number of peers is happening. 

Additionally some tests were broken as `discovery.stop` was happening just after `discovery.summary`, so I kept the same expected behaviour here.

Reply:
> moved SendDiscoverySummary where this change on the number of peers is happening.

But it does not happen here. It happens exactly when we receive an event from `events` channel. This block is called after `p.opts.TopicStopSearchDelay` which may be arbitrary. Sending this signal after a few seconds since adding a peer does not make sense and it's pretty confusing.